packages/regl-worldview/src/commands/Lines.js

```diff
@@ -346,7 +368,96 @@ const lines = (regl: any) => {
       pointArray.copyWithin(0, 3, 6);
       pointArray.copyWithin(n - 3, n - 6, n - 3);
     }
-  };
+  }
+
+  function fillPoseArrays(instances: number, poses: ?(Pose[]), hasInstancedPoses: boolean): ?Error {
+    if (hasInstancedPoses && poses && instances !== poses.length) {
+      return new Error(`Expected ${instances} poses but given ${poses.length} poses: will result in webgl error.`);
+    }
+    if (hasInstancedPoses && poses) {
+      if (positionArray.length < instances * 3) {
+        positionArray = new Float32Array(instances * 3);
+        rotationArray = new Float32Array(instances * 4);
+      }
+      for (let index = 0; index < poses.length; index++) {
+        const positionOffset = index * 3;
+        const rotationOffset = index * 4;
+        const { position, orientation: r } = poses[index];
+        const convertedPosition = Array.isArray(position) ? position : pointToVec3(position);
+        positionArray[positionOffset + 0] = convertedPosition[0];
+        positionArray[positionOffset + 1] = convertedPosition[1];
+        positionArray[positionOffset + 2] = convertedPosition[2];
+
+        const convertedRotation = Array.isArray(r) ? r : [r.x, r.y, r.z, r.w];
+        rotationArray[rotationOffset + 0] = convertedRotation[0];
+        rotationArray[rotationOffset + 1] = convertedRotation[1];
+        rotationArray[rotationOffset + 2] = convertedRotation[2];
+        rotationArray[rotationOffset + 3] = convertedRotation[3];
+      }
+    } else {
+      // If we don't have instanced poses, just load in the identity position / rotation vectors.
+      if (positionArray.length < VERTICES_PER_INSTANCE * 3) {
+        positionArray = new Float32Array(VERTICES_PER_INSTANCE * 3);
+        rotationArray = new Float32Array(VERTICES_PER_INSTANCE * 4);
+      }
+      positionArray.fill(0);
+      rotationArray.fill(0);
+      // Rotation array identity is [x: 0, y: 0, z: 0, w: 1]
+      for (let index = 0; index < VERTICES_PER_INSTANCE; index++) {
+        rotationArray[index * 4 + 3] = 1;
+      }
+    }
```

Comment:
You could statically allocate the non-instanced posePositionBuffer and poseRotationBuffer for one the command does not have instanced poses.  I think that's the happy path / default cause & in that situation instead of doing any of this work you could just pass the non-instanced buffers [right here](https://github.com/cruise-automation/webviz/pull/246/files#diff-60b9c6ba8357cc8cd56a2306263d0644R318-R325)